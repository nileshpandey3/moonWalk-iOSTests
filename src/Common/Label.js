import React from "react";
import styled from "styled-components";
import LinearGradient from "react-native-linear-gradient";

const Wrapper = styled(LinearGradient)`
  border-radius: 10px;
  padding: 10px 15px;
  margin: 0 10px;
`;

const LabelText = styled.Text`
  color: #8183a5;
  font-weight: bold;
`;

export default ({ text, numberOfLines = 1 }) => (
  <Wrapper
    colors={["#3a3a63", "#2c2f52"]}
    start={{ x: 0.0, y: 0.0 }}
    end={{ x: 1.0, y: 0.0 }}
  >
    <LabelText numberOfLines={numberOfLines}>{text}</LabelText>
  </Wrapper>
);
