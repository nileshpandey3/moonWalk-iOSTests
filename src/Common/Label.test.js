import React from "react";
import Label from "./Label";
import renderer from "react-test-renderer";

describe("label", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Label />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with text correctly", () => {
    const tree = renderer.create(<Label text="Label text" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
