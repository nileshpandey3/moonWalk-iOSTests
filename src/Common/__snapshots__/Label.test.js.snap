// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4282006115,
      4281085778,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 10,
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#8183a5",
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  />
</BVLinearGradient>
`;

exports[`label renders with text correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4282006115,
      4281085778,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 10,
        "marginBottom": 0,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#8183a5",
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Label text
  </Text>
</BVLinearGradient>
`;
